### PR TITLE
[WIP] Implementation of RBF mesh deformation method 

### PR DESCRIPTION
**Pull request by lucaabergo**
## Proposed Changes
I have been implementing for SU2 the radial basis function mesh deformation method with the selection of the control points using a greedy algorithm and a volume reduction technique to reduce the computational cost. The method can be used inside the shape optimization framework when calling SU2_DEF, keeping the standard ELA differentiated by codipack for the mesh sensitivity. 
References:
Abergo, Luca, Myles Morelli, and Alberto Guardone. "Aerodynamic Shape Optimization Based on Discrete Adjoint and RBF." Journal of Computational Physics (2023): 111951.

M. Morelli, T. Bellosta, A. Guardone, Efficient radial basis function mesh deformation methods for aircraft icing, J. Comput. Appl. Math. (2021) 113492,

## Related Work
*Resolve any issues (bug fix or feature request), note any related PRs, or mention interactions with the work of others, if any.*

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [ ] I am submitting my contribution to the develop branch.
- [ ] My contribution generates no new compiler warnings (try with --warnlevel=3 when using meson).
- [ ] My contribution is commented and consistent with SU2 style (https://su2code.github.io/docs_v7/Style-Guide/).
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp), if necessary.

### CODE REVIEW

**Review by pcarruscag — COMMENTED** [review]
Please remove the .bak files and rebase the branch or make a new branch and new PR.
Try not to commit copies of large files please because this stays in the history of the repo since we do not squash PRs

**Comment by NAnand-TUD** [comment]
@lucaabergo Currently I only see the config options in the PR, I guess the source code of implementation is coming later? Or are you planning to interact with an external code?

File: Common/src/CConfig.cpp [hunk 1]
```diff
@@ -2371,7 +2371,7 @@ void CConfig::SetConfig_Options() {
   addDoubleOption("KNOWLES_B", Knowles_B, 1.0);
   /* DESCRIPTION: Knowles N constant */
   addDoubleOption("KNOWLES_N", Knowles_N, 1.0);
-
+
```
Comment (WallyMaier) [inline Common/src/CConfig.cpp:2374]: Be sure to run clang format before pushing in

Comment (pcarruscag) [reply Common/src/CConfig.cpp:2374]: We don't clang-format the config yet...

**Comment by kursatyurt** [comment]
Hi, @lucaabergo is there any update? This sound interesting as I am currently working on morphing camber simulations.

**Comment by kursatyurt** [comment]
@tbellosta  may you have a contact with @lucaabergo? If he has no time, I can pull the implementation to the current version. Some fork/branch as you directed me for the rotor controls case would be nice.

**Comment by pcarruscag** [comment]
A PR was just opened that seems to contain more progress on this topic.

**Comment by pcarruscag** [comment]
I'll prune this branch if there are no updates

**Pull request closed without merging**